src/core/algorithms/algorithm.h

```diff
@@ -104,6 +104,14 @@ class Algorithm {
     std::type_index GetTypeIndex(std::string_view option_name) const;
 
     [[nodiscard]] std::unordered_set<std::string_view> GetPossibleOptions() const;
+
+    std::unordered_map<std::string_view, boost::any> GetOptValues() {
+        std::unordered_map<std::string_view, boost::any> opt_values;
+        for (auto& i : possible_options_) {
+            opt_values[i.second->GetName()] = i.second->GetOptValue();
```

Comment:
Check if the option is set before adding

Reply:
Do you really have to store all the options in a map? Isn't it possible to query the algorithm from Python one-by-one? I think that will be easier.

Reply:
I think it's better to return map objects because it's not clear what a function asking for options one by one should return if the requested option is missing or the option is not set.